service/src/main/java/bio/terra/cbas/dependencies/sam/SamService.java

```diff
@@ -28,4 +36,13 @@ public Result checkHealth() {
       return new Result(false, e.getMessage());
     }
   }
+
+  public UserStatusInfo getUserStatusInfo(String accessToken) throws InterruptedException {
+    UsersApi usersApi = samUsersApi(accessToken);
+    try {
+      return SamRetry.retry(usersApi::getUserStatusInfo);
+    } catch (ApiException apiException) {
+      throw SamExceptionFactory.create("Error getting user status info from Sam", apiException);
```

Comment:
Is this what's creating the `InterruptedException`? Is there a reason not to just let this throw the `ApiException` directly?

Reply:
Copied from https://github.com/DataBiosphere/terra-workspace-manager/blob/144c9c0aa4df21fdefd11f77f00b9df7f476a339/service/src/main/java/bio/terra/workspace/service/iam/SamService.java#L191-L200. When I dug into SamExceptionFactory I found this:
```
/**
 * Factory class for creating common Sam exceptions. This class pulls the error message out of the
 * Sam ApiException, which could be buried a layer down, and then inspects the HTTP Status code to
 * generate the right type of exception.
 */
```

Reply:
This method now handles the InterruptException by rethrowing the SamException version for consistency's sake, so the only type of exception coming out of this method will be an [`ErrorReportException`](https://github.com/DataBiosphere/terra-common-lib/blob/eaaf6217ec0f024afa45aac14d21c8964c0f27c5/src/main/java/bio/terra/common/exception/ErrorReportException.java#L15)